metisp/pymetis/src/pymetis/recipes/hci/metis_lm_adi_app.py

```diff
@@ -0,0 +1,161 @@
+#from pymetis.dataitems.hci import LmOffAxisPsfRaw, LmOnAxisPsfTemplate
+from pymetis.dataitems.hci.hci import LmAppCalibrated
+from pymetis.dataitems.hci.hci import AdiCalibrated
+
+
+from pymetis.dataitems.hci.hci import AdiCalibrated, LmAppSciCentred, LmAppCentroidTab
+from pymetis.dataitems.hci.hci import LmAppSciSpeckle, LmAppSciHifilt, LmAppSciDerotatedPsfsub
+from pymetis.dataitems.hci.hci import LmAppSciDerotated
+from pymetis.dataitems.hci.hci import LmAppSciContrastRadprof, LmAppSciContrastAdi, LmAppSciThroughput
+from pymetis.dataitems.hci.hci import LmAppSciCoverage, LmAppSciSnr, LmAppPsfMedian
+from pymetis.classes.recipes import MetisRecipe
+from pymetis.classes.prefab import RawImageProcessor
+from pymetis.classes.inputs import RawInput
+from pymetis.utils.dummy import create_dummy_header, create_dummy_image, create_dummy_table
+
+
+class MetisLmAppSciCalibrateImpl(RawImageProcessor):
+    class InputSet(RawImageProcessor.InputSet):
+        class RawInput(RawInput):
+            Item = LmSciCalibrated
+        #class LmOffAxisPsfRaw(RawInput):
+        #    Item = OffAxisPsf
+        #class LmOnAxisPsfTemplate(RawInput):
+        #    Item = OnAxisPsfTemplate
+
+        
+
+    ProductLmSciCalibrated = LmAppCalibrated
+    ProductLmSciCentred = LmAppSciCentred
+    ProductLmCentroidTab = LmAppCentroidTab
+    ProductLmSciSpeckle = LmAppSciSpeckle
+    ProductLmSciHifilt = LmAppSciHifilt
+    ProductLmSciDerotatedPsfsub = LmAppSciDerotatedPsfsub
+    ProductLmSciDerotated = LmAppSciDerotated
+    ProductLmSciContrastRadprof = LmAppSciContrastRadprof
+    ProductLmSciContrastAdi = LmAppSciContrastAdi
+    ProductLmSciThroughput = LmAppSciThroughput
+    ProductLmSciCoverage = LmAppSciCoverage
+    ProductLmSciSnr = LmAppSciSnr
+    ProductLmSciPsfMedian = LmAppPsfMedian
+    
+    def process(self) -> set[DataItem]:
+        
+            image = self.inputset.raw.load_data('DET1.DATA')[0]
+            #image = create_dummy_image()
+            table = create_dummy_table()
+
+            primary_header = create_dummy_header()
+            header_lmSciCalibrated = create_dummy_header()
+            header_lmSciCentred = create_dummy_header()
+            header_lmCentroidTable = create_dummy_header()
+            header_lmSciSpeckle = create_dummy_header()
+            header_lmSciHifilt = create_dummy_header()
+            header_lmSciDerotatedPsfsub = create_dummy_header()
+            header_lmSciDerotated = create_dummy_header()
+            header_lmSciContrastRadprof = create_dummy_header()
+            header_lmSciContrastAdi = create_dummy_header()
+            header_lmSciThroughput = create_dummy_header()
+            header_lmSciCoverage = create_dummy_header()
+            header_lmSciSnr = create_dummy_header()
+            header_lmSciPsfMedian = create_dummy_header()
+        
+            
+            product_lmSciCalibrated = self.ProductLmSciCalibrated(
+                    primary_header,
```

Comment:
Same thing here, you need to do something like
```suggestion
                    copy.deepcopy(primary_header),
```